### ISSUE TITLE
`LivePreviewState` naming collision

### ISSUE THREAD

**Issue by mnaoumov** [obsidianmd/obsidian-developer-docs#149]
There are two pages

1. https://github.com/obsidianmd/obsidian-developer-docs/blob/cc385fedfa8ba84a8909c71850b26b1b4e277616/en/Reference/TypeScript%20API/LivePreviewState.md
2. https://github.com/obsidianmd/obsidian-developer-docs/blob/cc385fedfa8ba84a8909c71850b26b1b4e277616/en/Reference/TypeScript%20API/livePreviewState.md

that differ only in casing.

This is going to be an issue on case-insensitive file systems.

**Comment by lishid** [comment]
Yeah that one has been annoying. I will find a better name for the interface.

**Comment by revolter** [comment]
Just stumbled upon this issue, and was confused not being able to do some rebases.

**Comment by revolter** [comment]
Did you find a better name? I could try to create a PR if I know what name to use.

**Comment by joethei** [comment]
We haven't found a better name yet, and we can't accept PR's to the relevant docs, as they are generated from our source code.

**Comment by mnaoumov** [comment]
How about `livePreviewStatePlugin` ?

**Comment by revolter** [comment]
> We haven't found a better name yet, and we can't accept PR's to the relevant docs, as they are generated from our source code.

I know. I was asking in order to create a PR for https://github.com/obsidianmd/obsidian-api.

**Comment by mnaoumov** [comment]
@revolter they don't accept PRs there either. It's all generated

**Comment by revolter** [comment]
Oooh, I see.

**Comment by lishid** [comment]
We can't rename the plugin name unfortunately because plugins are already using it. However we can change the type name and that 1) won't break existing plugins, and 2) will most likely not cause plugin build issues if the plugin is simply using the object. It is only a problem if a plugin explicitly specifies a variable of the return value, which is in most cases unnecessary.

I am considering updating it to `LPState` or `LivePreviewStateType`.

**Comment by mnaoumov** [comment]
@lishid you could consider instead of renaming the type, just modify the docs generator to generate unambiguous doc file name such as `LivePreviewState2.md`

**Comment by lishid** [comment]
The docs generator is unfortunately a gigantic monstrosity that I dare not attempt to try to modify. :)

**Comment by mnaoumov** [comment]
That's a pity as it has some serious flaws that you would need to address sooner or later anyways. I try to raise those flaws here and in the forum when I discover them

**Comment by revolter** [comment]
Any updates after ~2 months?

**Comment by lishid** [comment]
Fixed by https://github.com/obsidianmd/obsidian-api/commit/165ccdda6d2b0f1bbed890f94f07facd91ad5da5
